text/0125-xcm-asset-metadata.md

```diff
@@ -0,0 +1,233 @@
+        asset_id: AssetId,
+
+        /// The ID of an asset instance.
+        ///
+        /// If the value is `Undefined`, the reported metadata is related to the collection, not a token.
+        instance: AssetInstance,
+
+        /// See `MetadataResponseData` below.
+        data: MetadataResponseData,
+    }
+}
+
+pub enum MetadataResponseData {
+    /// The metadata key list to be reported
+    /// in response to the `KeySet` metadata query kind.
+    KeySet(MetadataKeySet),
+
+    /// The values of the keys that were specified in the
+    /// `Values` variant of the metadata query kind.
+    Values(MetadataMap),
+}
+```
+
+#### `ModifyMetadata`
+
+The `ModifyMetadata` is a new instruction to request a remote chain to modify the values of the specified keys.
+
+This instruction can be used to update the metadata of a collection (fungible or nonfungible) or of an NFT.
+
+The remote chain handles the modification request and may reject it based on its internal rules.
+The request can only be executed or rejected in its entirety. It must not be executed partially.
+
+To execute the `ModifyMetadata`, an origin is required so that the handling logic can authorize the metadata modification request from a known source. Since this instruction requires an origin, the assets used to cover the execution fees must be transferred in a way that preserves the origin. For instance, one can use the approach described in RFC #122 if the handling chain configured aliasing rules accordingly.
+
+The example use case of this instruction is to ask the reserve location of the asset to modify the metadata. So that, the original asset's metadata is updated according to the reserve location's rules.
+
+```rust
+ModifyMetadata {
+    /// The ID of an asset (a collection, fungible or nonfungible).
+    asset_id: AssetId,
+
+    /// The ID of an asset instance.
+    ///
+    /// If the value is `Undefined`, the modification request targets the collection, not a token.
+    instance: AssetInstance,
+
+    /// The map contains the keys mapped to the requested new values.
+    modification: MetadataMap,
+}
+```
+
+### Repurposing `AssetInstance::Undefined`
+
+As the new instructions show, this RFC reframes the purpose of the `Undefined` variant of the `AssetInstance` enum.
+This RFC proposes to use the `Undefined` variant of a collection identified by an `AssetId` as a synonym of the collection itself. I.e., an asset `Asset { id: <AssetId>, fun: NonFungible(AssetInstance::Undefined) }` is considered an NFT representing the collection itself.
+
+As a singleton non-fungible instance is barely distinguishable from its collection, this convention shouldn't cause any problems. 
+
+Thus, the `AssetInstance` docs must be updated accordingly in the implementations.
+
+## Drawbacks
+
+Regarding ergonomics, no drawbacks were noticed.
+
+As for the user experience, it could discover new cross-chain use cases involving asset collections and NFTs, indicating a positive impact.
```

Comment:
I'm wondering if there are any metadata keys that we can make known and standard instead of leaving everything as a `Vec<u8>`. But I think this is a good start and cross-chain NFT standards can start to emerge from this.